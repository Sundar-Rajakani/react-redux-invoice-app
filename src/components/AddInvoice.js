import React, { Component } from 'react';
import { connect } from 'react-redux';

import { addInvoice } from '../actions/invoices'
import InvoiceForm from './InvoiceForm';


export class AddInvoice extends Component {

  onSubmit = (invoice) => {
    this.props.addInvoice(invoice);  
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <h1>Invoice Form</h1>
        <InvoiceForm
          onSubmit={this.onSubmit} />
      </div>
    )
  }
}


const mapDispatchToProps = (dispatch) => ({
  addInvoice: (invoice) => dispatch.addInvoice(invoice)
})
export default connect(undefined, mapDispatchToProps)(AddInvoice);
