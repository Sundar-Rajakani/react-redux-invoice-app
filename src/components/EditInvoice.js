import React, { Component } from 'react';
import { connect } from 'react-redux';

import { editInvoice, removeInvoice } from '../actions/invoices';
import InvoiceForm from './InvoiceForm';



export class EditInvoice extends Component {


  onSubmit = (invoice) => {
    this.props.editInvoice(props.invoice.id, invoice);
    this.props.history.push('/');
  }

  onClick = (e) => {
    this.props.removeInvoice({ id: props.invoice.id });
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <h1>Edit Invoice</h1>
        <InvoiceForm
          invoice={this.props.invoice}
          onSubmit={this.onSubmit} />
        <button onClick={this.onClick}>Remove</button>
      </div >
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    invoice: state.invoices.find((invoice) => invoice.id === props.match.params.id)
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editInvoice: (id, invoice) => dispatch(editInvoice(id, invoice)),
  removeInvoice: (data) => dispatch(removeInvoice(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditInvoice);
