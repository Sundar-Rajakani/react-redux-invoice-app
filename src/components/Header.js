import React from 'react';
import { NavLink } from 'react-router-dom';
const Header = () => (
  <header>
    <h1>Invoice Management</h1>
    <NavLink to="/" activeClassName="isActive" exact={true}>Home</NavLink>
    <NavLink to="/add" activeClassName="isActive">AddInvoice</NavLink>
    <NavLink to="/help" activeClassName="isActive">Help</NavLink>
  </header>
);

export default Header;
