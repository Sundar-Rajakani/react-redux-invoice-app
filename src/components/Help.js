import React from 'react';


const Help = () => (
  <div>This is help hage</div>
);

export default Help;
