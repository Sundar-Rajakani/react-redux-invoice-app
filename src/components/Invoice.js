import React from 'react';
import { Link } from 'react-router-dom';

export const Invoice = ({ description, amount, createdAt, id, dispatch }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount} - {createdAt}</p>

  </div>
);

export default Invoice;
