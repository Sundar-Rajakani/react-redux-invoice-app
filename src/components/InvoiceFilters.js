import React, { Component } from 'react';
import { connect } from 'react-redux';

import { DateRangePicker } from 'react-dates';

import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters';

class InvoiceFilters extends Component {
  state = {
    calendarFocused: null
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  }

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  }

  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate(e.target.value)
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount(e.target.value)
    }
  }

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div >
    )
  }
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByAmount: () => dispatch(sortByAmount()),
  sortByDate: () => dispatch(sortByDate()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))

})

export default connect(mapStateToProps, mapDispatchToProps)(InvoiceFilters);
