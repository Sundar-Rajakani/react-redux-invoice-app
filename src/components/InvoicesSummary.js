import React, { Component } from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

import selectInvoicesTotal from '../selectors/invoices-total.js';
import getFilteredInvoices from '../selectors/invoices.js';

export const InvoicesSummary = ({ noOfInvoices, totalOfInvoices }) => {
  return (
    <div>
      <h1>Viewing {noOfInvoices} {noOfInvoices === 1 ? 'invoice' : 'invoices'} , with a total of {numeral(totalOfInvoices / 100).format('$0,0.00')}</h1>
    </div>
  )
}

const mapStateToProps = (state) => {
  const invoices = getFilteredInvoices(state.invoices, state.filters);
  return {
    noOfInvoices: invoices.length,
    totalOfInvoices: selectInvoicesTotal(invoices)
  }
}

export default connect(mapStateToProps)(InvoicesSummary);