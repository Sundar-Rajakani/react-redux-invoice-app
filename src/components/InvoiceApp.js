import React from 'react';

import Invoices from './Invoices';
import InvoicesSummary from './InvoicesSummary'
import InvoiceFilters from './InvoiceFilters';

const InvoiceApp = () => (
  <div>
    <InvoicesSummary />
    <InvoiceFilters />
    <Invoices />
  </div>
);

export default InvoiceApp;