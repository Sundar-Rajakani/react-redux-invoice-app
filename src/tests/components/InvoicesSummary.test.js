import React from 'react';
import { shallow } from 'enzyme';
import { InvoiceFilters } from '../../components/InvoiceFilters';
import { InvoicesSummary } from '../../components/InvoicesSummary';

test('shoudl render invoices summary for a single invoice', () => {
  const wrapper = shallow(<InvoicesSummary noOfInvoices={1} totalOfInvoices={235}/>);
  expect(wrapper).toMatchSnapshot();
})

test('shoudl render invoices summary for multiple invocies', () => {
  const wrapper = shallow(<InvoicesSummary noOfInvoices={10} totalOfInvoices={23587687} />);
  expect(wrapper).toMatchSnapshot();
})