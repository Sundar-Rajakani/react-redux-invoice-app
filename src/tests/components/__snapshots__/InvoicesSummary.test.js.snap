// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoudl render invoices summary for a single invoice 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    invoice
     , with a total of 
    $2.35
  </h1>
</div>
`;

exports[`shoudl render invoices summary for multiple invocies 1`] = `
<div>
  <h1>
    Viewing 
    10
     
    invoices
     , with a total of 
    $235,876.87
  </h1>
</div>
`;
